src/query/service/src/interpreters/util.rs

```diff
@@ -61,3 +84,300 @@ pub fn generate_desc_schema(
     }
     (names, types, nulls, default_exprs, extras)
 }
+
+#[derive(Debug, Clone)]
+pub struct QueryResult {
+    pub(crate) schema: DataSchemaRef,
+    pub(crate) block: DataBlock,
+}
+
+pub struct ScriptClient {
+    pub(crate) ctx: Arc<QueryContext>,
+}
+
+impl Client for ScriptClient {
+    type Var = Scalar;
+    type Set = QueryResult;
+
+    async fn query(&self, query: &str) -> databend_common_exception::Result<Self::Set> {
+        let ctx = self
+            .ctx
+            .get_current_session()
+            .create_query_context()
+            .await?;
+
+        let mut planner = Planner::new(ctx.clone());
+        let (plan, _) = planner.plan_sql(query).await?;
+        let interpreter = InterpreterFactory::get(ctx.clone(), &plan).await?;
+        let stream = interpreter.execute(ctx.clone()).await?;
+        let blocks = stream.try_collect::<Vec<_>>().await?;
+        let schema = plan.schema();
+
+        let block = match blocks.len() {
+            0 => DataBlock::empty_with_schema(schema.clone()),
+            1 => blocks[0].clone(),
+            _ => DataBlock::concat(&blocks)?,
+        };
+
+        Ok(QueryResult { schema, block })
+    }
+
+    fn var_to_ast(&self, scalar: &Self::Var) -> databend_common_exception::Result<Expr> {
+        let ast = match scalar {
```

Comment:
we can use `scalar.sql_display` and parse it back

Reply:
It's refactored from src/query/service/src/interpreters/interpreter_execute_immediate.rs. cc @andylokandy Cloud we replace it as sql_display?

Reply:
May use `ScalarRef::to_string` and then `parse_expr` it back. But, currently, it's not guaranteed that it must success.

Reply:
`sql_display`  handles inf/nan better.

Reply:
see `parse_computed_expr_to_string`

Reply:
Done